### PR TITLE
[Rest Api Compatibility] MovAvgPipelineAggregation

### PR DESCRIPTION
**Pull request by joegallo**
Related to #51816 / #68905.

The `moving_avg` pipeline aggregation was deprecated in https://github.com/elastic/elasticsearch/pull/29594 (6.4.0+) and removed via https://github.com/elastic/elasticsearch/pull/39328, this PR adds back a nicer error message (but only in the presence of the relevant `compatible-with=7` headers).

Quick action shot:
```
> GET /twitter-1/_search?pretty HTTP/1.1
> Host: localhost:9200
> User-Agent: curl/7.64.1
> Accept: application/vnd.elasticsearch+json;compatible-with=7
> Content-Type: application/vnd.elasticsearch+json;compatible-with=7
> Content-Length: 338
>
* upload completely sent off: 338 out of 338 bytes
< HTTP/1.1 400 Bad Request
< X-elastic-product: Elasticsearch
< Warning: 299 Elasticsearch-8.1.0-SNAPSHOT-bc6bceaca3bbfacfcc92b8a2480db5d38e1c5dec "Deprecated field [moving_avg] used, replaced by [Moving Average aggregation usage is not supported. Use the [moving_fn] aggregation instead.]"
< Warning: 299 Elasticsearch-8.1.0-SNAPSHOT-bc6bceaca3bbfacfcc92b8a2480db5d38e1c5dec "Moving Average aggregation usage is not supported. Use the [moving_fn] aggregation instead."
< content-type: application/json;charset=utf-8
< content-length: 452
<
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parsing_exception",
        "reason" : "Moving Average aggregation usage is not supported. Use the [moving_fn] aggregation instead.",
        "line" : 14,
        "col" : 25
      }
    ],
    "type" : "parsing_exception",
    "reason" : "Moving Average aggregation usage is not supported. Use the [moving_fn] aggregation instead.",
    "line" : 14,
    "col" : 25
  },
  "status" : 400
}
* Connection #0 to host localhost left intact
* Closing connection 0
```

I expect that the vast majority of direct users of `moving_avg` have migrated over the last ~3.5 years of it being deprecated, and if you were using it indirectly via Kibana's Moving Avg then you've already been migrated to `moving_fn` for quite some time (see https://github.com/elastic/kibana/issues/24702).

Side note: funnily enough, because `moving_avg` and `moving_fn` are so close together in terms of edit distance, 8.0.0 without the compat header still **suggests** the latter if you present it with the former. You still get an error, it's the same error you'd get for a typo like `movnig_fn`.

### CODE REVIEW

**Review by jakelandis — COMMENTED** [review]
Changes look good on the REST compat front, @not-napoleon - can you review the agg portion ?

one nit: can you add a comment to the Builder that it can be removed once support for compatibility with v7 is removed.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.elasticsearch.core.RestApiVersion;
+import org.elasticsearch.index.query.CommonTermsQueryBuilder;
+import org.elasticsearch.xcontent.ContextParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class MovAvgPipelineAggregationBuilder extends AbstractPipelineAggregationBuilder<MovAvgPipelineAggregationBuilder> {
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java:24]: Should we also `@deprecate` this?

**Review by not-napoleon — APPROVED** [review]
For an aggregation that's intended to fail at parse time, this looks fine.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java (hunk 1 above)
Comment (joegallo) [reply server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java:33]: 👍, c1d7a35

**Pull request merged**